.github/workflows/windows_build.yml

```diff
@@ -1,37 +1,73 @@
-name: Windows-Build
+name: Windows-2019
 
 on:
   push:
     branches:
-      - master
+      - main
   pull_request:
   schedule:
     - cron: '0 5 * * *'
 
+env:
+  VCPKG_PKGS: >-
+    boost-graph eigen3 console-bridge
+
 jobs:
   windows_ci:
-    name: Noetic
+    name: Windows-2019
     runs-on: windows-2019
     steps:
-    - uses: actions/checkout@v2
+    - name: Checkout repository
+      uses: actions/checkout@master
       with:
-        submodules: recursive
         path: src/descartes_light
-    - name: Build
+
+    - name: checkout-vcpkg
+      uses: actions/checkout@master
+      with:
+        path: vcpkg
+        repository: microsoft/vcpkg
+
+    - name: bootstrap-vcpkg
+      working-directory: vcpkg
+      run: bootstrap-vcpkg.bat
+
+    - name: vcpkg-dry-run
+      working-directory: vcpkg
       shell: cmd
       run: |
-        set ChocolateyInstall=C:\opt\chocolatey
-        choco sources add -n=ros-win -s="https://aka.ms/ros/public" --priority=1
+        vcpkg install --dry-run --triplet x64-windows ${{ env.VCPKG_PKGS }} > vcpkg_dry_run.txt
 
-        call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\Common7\Tools\VsDevCmd.bat" -arch=amd64 -host_arch=amd64
+    - name: cache-vcpkg-archives
+      if: startsWith(github.ref, 'refs/tags/v') != true
+      id: cache-vcpkg-archives
+      uses: pat-s/always-upload-cache@v3
+      with:
+        path: C:\Users\runneradmin\AppData\Local\vcpkg\archives
+        key: ${{ runner.os }}-x64-vcpkg-${{ hashFiles('vcpkg/vcpkg_dry_run.txt') }}-main
 
+    - name: install-depends
+      shell: cmd
+      run: |
+        vcpkg integrate install
         python -m pip install vcstool -q
         python -m pip install colcon-common-extensions -q
-        choco install console_bridge -y --no-progress
-        choco install boost -y --no-progress
-        choco install eigen -y --no-progress
-        refreshenv
+        vcpkg install --triplet x64-windows ${{ env.VCPKG_PKGS }}
 
-        vcs import --input "${{ github.workspace }}/src/descartes_light/dependencies.rosinstall" src/
+    - name: configure-msvc
+      uses: ilammy/msvc-dev-cmd@master
+      with:
+        arch: x64
 
-        colcon build --cmake-args -DCMAKE_BUILD_TYPE=Release -DDESCARTES_ENABLE_TESTING=ON -DCMAKE_PREFIX_PATH=C:\opt\rosdeps
+    - name: build-workspace
+      shell: cmd
```

Comment:
If you make the recommend change above you will need to add `working-directory: workspace` to this and the the run-test below.